tool/tctl/common/access_request_command.go

```diff
@@ -278,10 +280,44 @@ func (c *AccessRequestCommand) Create(client auth.ClientI) error {
 }
 
 func (c *AccessRequestCommand) Delete(client auth.ClientI) error {
+	ctx := context.TODO()
+	var approvedTokens []string
 	for _, reqID := range strings.Split(c.reqIDs, ",") {
-		if err := client.DeleteAccessRequest(context.TODO(), reqID); err != nil {
+		// Fetch the requests first to see if they were approved to provide the
+		// proper messaging.
+		reqs, err := client.GetAccessRequests(ctx, types.AccessRequestFilter{
+			ID: reqID,
+		})
+		if err != nil {
 			return trace.Wrap(err)
 		}
+		if len(reqs) != 1 {
+			return trace.BadParameter("request with ID %q not found", reqID)
+		}
+		for _, req := range reqs {
```

Comment:
You don't need a loop here, you can just check `reqs[0]` since you already know that `len(reqs)` is exactly 1.

Reply:
good call, done